crates/agglayer-prover/src/executor.rs

```diff
@@ -33,8 +33,8 @@ pub(crate) const ELF: &[u8] =
 
 #[derive(Clone)]
 pub struct Executor {
-    network: Option<BoxCloneService<Request, Response, Error>>,
-    local: Option<BoxCloneService<Request, Response, Error>>,
+    primary: BoxCloneService<Request, Response, Error>,
+    fallback: Option<BoxCloneService<Request, Response, Error>>,
```

Comment:
Would it make sense for this to have an arbitrary number of fallbacks (including zero)? I.e. replace the `Option` with a sequence of some sort?

Reply:
We could. Do we need to at this time?

Reply:
> We could. Do we need to at this time? I mean would we realistically use that sequencer any time?

No, just a random suggestion. Just `Option` as a parameter is often a code smell that suggests a generalization is possible.

Reply:
In this case, feels cleaner/simpler to ask the user to set a simple fallback prover that exists/doesn't exist in the toml file than to create an array to say I want one fallback prover. Happy to add this sequence if we meet the case where we need it.